src/infra/core/src/repos/dataset_repository_local_fs.rs

```diff
@@ -336,12 +320,21 @@ impl DatasetRepository for DatasetRepositoryLocalFs {
     async fn create_dataset_from_snapshot(
         &self,
         snapshot: DatasetSnapshot,
+        publicly_available: bool,
```

Comment:
Let's introduce `CreateDatasetOpts` structure right away. I think public/private is just the first out of many parameters to come (e.g. storage backend selection), so its better to add `CreateDatasetOpts::default()` in all the bazillion places now and avoid such changes later.

Reply:
Yes, it's better to really use that pattern -- absolutely agree

I added the option structure, but within the use cases (I decided to move it there, since after the introduction of usecase as a separate entity, it is better to have additional logic there)